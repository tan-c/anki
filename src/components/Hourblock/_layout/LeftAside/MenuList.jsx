import React from 'react';
import PropTypes from 'prop-types';
// import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';

import { Link, withRouter } from 'react-router-dom';

export class MenuList extends React.Component {
  // constructor(props, context) {
  //  super(props, context);
  // }

  // componentWillReceiveProps(nextProps) {
  //  this.setState({
  //   });
  // }

  render() {
    const { location } = this.props;

    return (
      <nav id="menu" data-role="menu-list">
        <ul>
          <li className={location.pathname === '/hourblock' ? 'bg-green' : ''}>
            <Link to="/hourblock">
              <i className="fa fa-fw fa-dashboard" />
            </Link>
          </li>
          <li className={location.pathname === '/hourblock/planning' ? 'bg-green' : ''}>
            <Link to="/hourblock/planning">
              <i className="fa fa-fw fa-calendar" />
            </Link>
          </li>
          <li className={location.pathname === '/hourblock/tasks' ? 'bg-green' : ''}>
            <Link to="/hourblock/tasks">
              <i className="fa fa-fw fa-list" />
            </Link>
          </li>
          <li className={location.pathname === '/hourblock/settings' ? 'bg-green' : ''}>
            <Link to="/hourblock/settings">
              <i className="fa fa-fw fa-cog" />
            </Link>
          </li>
        </ul>
      </nav>
    );
  }
}

MenuList.propTypes = {
  location: PropTypes.object.isRequired,
};

// function mapStateToProps(state, ownProps) {
//   return {
//   };
// }
//
// function mapDispatchToProps(dispatch) {
//   return {
//   };
// }

export default withRouter(MenuList);
