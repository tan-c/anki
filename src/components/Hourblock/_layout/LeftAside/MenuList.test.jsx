import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { MenuList } from './MenuList';

function setup() {
  const props = {
    location: {},
  };

  return shallow(<MenuList {...props} />);
}

describe('MenuList Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find("nav[data-role='menu-list']")).toHaveLength(1);
    });
  });
});
