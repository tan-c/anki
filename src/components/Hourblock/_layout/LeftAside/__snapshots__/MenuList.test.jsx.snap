// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuList Unit Tests when normally set up renders correctly 1`] = `
<nav
  data-role="menu-list"
  id="menu"
>
  <ul>
    <li
      className=""
    >
      <Link
        replace={false}
        to="/"
      >
        <i
          className="fa fa-fw fa-dashboard"
        />
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        replace={false}
        to="/planning"
      >
        <i
          className="fa fa-fw fa-calendar"
        />
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        replace={false}
        to="/tasks"
      >
        <i
          className="fa fa-fw fa-list"
        />
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        replace={false}
        to="/settings"
      >
        <i
          className="fa fa-fw fa-cog"
        />
      </Link>
    </li>
  </ul>
</nav>
`;
